src/codegen/deleter.cpp

```diff
@@ -38,6 +38,7 @@ void Deleter::Delete(uint32_t tile_group_id, uint32_t tuple_offset) {
 
   auto *txn = executor_context_->GetTransaction();
   auto tile_group = table_->GetTileGroupById(tile_group_id);
+  PELOTON_ASSERT(tile_group != nullptr);
```

Comment:
Why was this line added? Is this necessary? Or was it for debugging? 
Can it cause undefined behavior in Release mode?

Reply:
GetTileGroupById can now return a nullptr, this is just a check that we don't hit that during make check and other Debug builds. That's what most of the ASSERTs added to the execution layer are looking for.

If you're suggesting that we make it impossible to return a nullptr to the execution level, then we need to revisit how we access TileGroups entirely and go only through an iterator that will always be valid.

Reply:
I don't think you need to touch the execution layer. In a correct implementation, the execution layer should be completely oblivious to the GC. 

_**GetTileGroupById can now return a nullptr if the TileGroup has been dropped.**_ -> That is my biggest concern. So this thing is checked in debug which is great. But what happens in release mode? If this function gets a nullptr, it hits a segfault and peloton crashes? That's not expected behavior.

Like I said, in my overall comment, the whole point of the epoch based garbage collection is that the `TileGroup` references are removed only when you are certain that no other transaction is accessing it.  This is very similar to GC in the SkipListIndex.  The storage may get these `nullptr` references, but the execution layer should never get such a reference. 

I would suggest that we can either do a hangouts call over the weekend. Or you talk to @apavlo about this.